### PR TITLE
Create getauthenticationstateasync-was-called-before-setauthenticatio…

### PR DESCRIPTION
**Pull request by nelci592**
…nstate.md

### CODE REVIEW

File: knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Description
+
+Error 'GetAuthenticationStateAsync was called before SetAuthenticationState.' is thrown in a [Custom Report Source Resolver](../reporting/telerik-reporting-rest-custom-report-resolver) that uses **GetAuthenticationStateAsync**.
+
+## Error Message
+GetAuthenticationStateAsync was called before SetAuthenticationState.
+
+## Cause\Possible Cause(s)
+We require Singleton for our service, however, Singleton can't be used because the AuthenticationProvider requires Scoped. You can find more information in [Accessing an authenticated user outside of a view in Blazor](https://stackoverflow.com/questions/59744356/accessinging-an-authenticated-user-outside-of-a-view-in-blazor) Stack Overflow thread.
```
Comment (DimitarNikolovv06) [inline knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md:32]: Maybe we could say here "We require the use of a Singleton for our service".

**Review by DimitarNikolovv06 — COMMENTED** [review]
It's looking pretty good.

File: knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+title: GetAuthenticationStateAsync was called before SetAuthenticationState
+description: The issue was resolved using the GetIdentity method of our controller. The error will be described in a KB article.  Error 'GetAuthenticationStateAsync was called before SetAuthenticationState.' is thrown in a custom report source resolver that uses GetAuthenticationStateAsync
```
Comment (todorarabadzhiev) [inline knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md:3]: I think you should re-write this description

File: knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Error 'GetAuthenticationStateAsync was called before SetAuthenticationState.' is thrown in a [Custom Report Source Resolver](../reporting/telerik-reporting-rest-custom-report-resolver) that uses **GetAuthenticationStateAsync**.
+
+## Error Message
+GetAuthenticationStateAsync was called before SetAuthenticationState.
+
+## Cause\Possible Cause(s)
+We require the usage Singleton for our service, however, Singleton can't be used because the AuthenticationProvider requires Scoped. You can find more information in [Accessing an authenticated user outside of a view in Blazor](https://stackoverflow.com/questions/59744356/accessinging-an-authenticated-user-outside-of-a-view-in-blazor) Stack Overflow thread.
+
+## Solution
+We support accessing the current user inside the CustomReportSourceResolver using our own mechanism as described in [ASP.NET Core. How to use information from HttpContext in Custom Report Resolver](./core-how-to-pass-information-from-httpcontext-to-reporting-engine). The correct user and claims are passed to the GetIdentity() method. Although that not all dependencies are passed using dependency injection, note that this is only a proof that the approach works.
```
Comment (todorarabadzhiev) [inline knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md:35]: I don't understand the last sentence 'Although that not all dependencies are passed using dependency injection, note that this is only a proof that the approach works. '

**Review by todorarabadzhiev — CHANGES_REQUESTED** [review]
There are things that should be clarified

File: knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+Error 'GetAuthenticationStateAsync was called before SetAuthenticationState.' is thrown in a [Custom Report Source Resolver](../reporting/telerik-reporting-rest-custom-report-resolver) that uses **GetAuthenticationStateAsync**.
```
Comment (todorarabadzhiev) [inline knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md:26]: Use SLUG 
{%slug telerikreporting/using-reports-in-applications/host-the-report-engine-remotely/telerik-reporting-rest-services/rest-service-report-source-resolver/how-to-implement-a-custom-report-source-resolver%}

File: knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+## Error Message
+GetAuthenticationStateAsync was called before SetAuthenticationState.
+
+## Cause\Possible Cause(s)
+The Telerik Reporting REST Service is recommended to be in a Singleton scope. However, the AuthenticationProvider requires utilizing the Scoped option. You can find more information in [Accessing an authenticated user outside of a view in Blazor](https://stackoverflow.com/questions/59744356/accessinging-an-authenticated-user-outside-of-a-view-in-blazor) Stack Overflow thread.
+
+## Solution
+We support accessing the current user inside the CustomReportSourceResolver using our own mechanism as described in [ASP.NET Core. How to use information from HttpContext in Custom Report Resolver](./core-how-to-pass-information-from-httpcontext-to-reporting-engine). For example:
```
Comment (todorarabadzhiev) [inline knowledge-base/getauthenticationstateasync-was-called-before-setauthenticationstate.md:35]: Use SLUG
{%slug core-how-to-pass-information-from-httpcontext-to-reporting-engine%}

**Review by todorarabadzhiev — APPROVED** [review]

**Pull request merged**